In [1]:
import POLIST_00_graph_analysis as ga
import pandas as pd
import numpy as np

pd.set_option("display.max_columns", None)

datas = pd.read_csv("POLIST_13_kmean_datas.csv")


In [2]:
from sklearn.preprocessing import OneHotEncoder

X = datas.customer_state.to_numpy().reshape(-1, 1)
ohe = OneHotEncoder().fit(X)

X_ohe = pd.DataFrame(ohe.transform(X).toarray(), columns= ohe.get_feature_names_out(["customer_state__"]), index= datas.index)

In [3]:
datas.drop("customer_state", axis= 1, inplace= True)

datas = datas.merge(X_ohe, left_index= True, right_index= True)
datas

customer_unique_id  nb_score_sum  pay_sum_sum  \
0      7c396fd4830fd04220f754e42b4e5bff             2        82.82   
1      af07308b275d755c9edb36a90c618231             1       141.46   
2      3a653a41f6f9fc3d2a113cf8398680e8             1       179.12   
3      7c142cf63193a1473d2e66489a9ae977             1        72.20   
4      72632f0f9dd73dfee390c9b22eb56dd6             1        28.62   
...                                 ...           ...          ...   
92826  6359f309b166b0196dbf7ad2ac62bb5a             1        85.08   
92827  da62f9e57a76d978d02ab5362c509660             1       195.00   
92828  737520a9aad80b3fbbdad19b66b37b30             1       271.01   
92829  5097a5312c8b157bb7be58ae360ef43c             1       441.16   
92830  60350aa974b26ff12caad89e55993bd6             1        86.86   

       nb_product_sum_sum  type_boleto_sum  type_credit_card_sum  \
0                       2              0.0                   2.0   
1                       1              1.0                   0.0   
2                       1              0.0                   1.0   
3                       1              0.0                   1.0   
4                       1              0.0                   1.0   
...                   ...              ...                   ...   
92826                   1              0.0                   1.0   
92827                   1              0.0                   1.0   
92828                   1              0.0                   1.0   
92829                   2              0.0                   1.0   
92830                   1              0.0                   0.0   

       type_debit_card_sum  type_voucher_sum  \
0                      0.0               2.0   
1                      0.0               0.0   
2                      0.0               0.0   
3                      0.0               0.0   
4                      0.0               0.0   
...                    ...               ...   
92826                  0.0               0.0   
92827                  0.0               0.0   
92828                  0.0               0.0   
92829                  0.0               0.0   
92830                  1.0               0.0   

       category__agro_industry_and_commerce_sum  \
0                                           0.0   
1                                           0.0   
2                                           0.0   
3                                           0.0   
4                                           0.0   
...                                         ...   
92826                                       0.0   
92827                                       0.0   
92828                                       0.0   
92829                                       0.0   
92830                                       0.0   

       category__air_conditioning_sum  category__art_sum  \
0                                 0.0                0.0   
1                                 0.0                0.0   
2                                 0.0                0.0   
3                                 0.0                0.0   
4                                 0.0                0.0   
...                               ...                ...   
92826                             0.0                0.0   
92827                             0.0                0.0   
92828                             0.0                0.0   
92829                             0.0                0.0   
92830                             0.0                0.0   

       category__arts_and_craftmanship_sum  category__audio_sum  \
0                                      0.0                  0.0   
1                                      0.0                  0.0   
2                                      0.0                  0.0   
3                                      0.0                  0.0   
4                                      0.0                  0.0   
...                                    ...        

In [4]:
from POLIST_00_graph_analysis import segmentation

segment_names = ["cluster " + str(x) for x in np.sort(datas["labels"].unique())]
conditions = [datas['labels'] == x for x in np.sort(datas["labels"].unique())]

segments = segmentation(datas, conditions, segment_names)
segments

segment_name  nb_clients  perc_clients  payment_mean_value  montant_CA  \
0    cluster 0        3183           3.4              665.13  2117122.85   
1    cluster 1       28798          31.0               95.05  2737237.65   
2    cluster 2         244           0.3             2429.46   592787.97   
3    cluster 3       10280          11.1              309.54  3182039.11   
4    cluster 4       29520          31.8              101.95  3009604.59   
5    cluster 5       19878          21.4              120.77  2400743.68   
6    cluster 6           7           0.0             7881.47    55170.31   
7    cluster 7         921           1.0             1283.11  1181746.28   

   perc_CA  buyed_prod_mean  prod_price_mean  buyed_prod_nb  \
0     13.9             1.55           429.12           4920   
1     17.9             1.10            86.41          31780   
2      3.9             1.77          1372.58            432   
3     20.8             1.43           216.46          14690   
4     19.7             1.11            91.85          32789   
5     15.7             1.14           105.94          22586   
6      0.4             3.29          2395.58             23   
7      7.7             1.63           787.18           1499   

   perc_buyed_prod_nb  perc_boleto_payment  perc_credit_card_payment  \
0                 4.5                 15.7                      78.4   
1                29.2                 20.5                      72.7   
2                 0.4                 18.2                      77.7   
3                13.5                 15.6                      79.3   
4                30.2                 18.5                      74.3   
5                20.8                 20.7                      71.3   
6                 0.0                 50.0                      50.0   
7                 1.4                 15.3                      78.9   

   perc_debit_card_payment  perc_voucher_payment             best_cat_1  \
0                      0.9                   5.0          watches gifts   
1                      1.2                   5.6         bed bath table   
2                      1.9                   2.2  computers accessories   
3                      1.1                   4.0         bed bath table   
4                      2.4                   4.8          health beauty   
5                      0.9                   7.2         bed bath table   
6                      0.0                   0.0        fixed telephony   
7                      0.8                   5.0          watches gifts   

   perc_best_cat_1           best_cat_2  perc_best_cat_2  \
0             13.0        health beauty             10.0   
1             10.3       sports leisure              8.4   
2             10.1  musical instruments              9.4   
3              9.7        health beauty              9.1   
4             10.4       bed bath table              9.9   
5             10.6      furniture decor              8.3   
6             25.0     small appliances             25.0   
7             11.5        health beauty              9.3   

                   best_cat_3  perc_best_cat_3  \
0              bed bath table              6.5   
1               health beauty              8.0   
2                        auto              9.0   
3               watches gifts              8.3   
4                  housewares              7.2   
5              sports leisure              8.0   
6  agro industry and commerce             12.5   
7                   computers              8.5   

                          worst_cat_1  perc_worst_cat_1  \
0                      home comfort 2               0.0   
1                           computers               0.0   
2                                food               0.0   
3                      home comfort 2               0.0   
4                          la cuisine               0.0   
5  kitchen laptops and food preparers               0.0   
6                  

### Détails

1. Le ***cluster 0*** représentent 3.4% des clients. En moyenne, ils achètent 1.55 articles au prix moyen de 429.12, soit un panier moyen de 665.13. Ils participent à hauteur de 13.9% du CA total et payent majoritairement par carte de crédit (78.4%) et par boleto (15.7%). Leurs catégories préférées sont "watches gifts" (13.0%), "health beauty" (10.0%) et "bed bath table" (6.5%). Ces clients ont passé en moyenne 1.1 commandes dont la dernière a eu lieu il y a en moyenne 249.3 jours à la date de l'analyse. Ces clients proviennent à 34.7% de l'état "SP". Ils ont donné un score moyen de 4.0 étoiles.
2. Le ***cluster 1*** représentent 31.0% des clients. En moyenne, ils achètent 1.10 articles au prix moyen de 86.41, soit un panier moyen de 95.05. Ils participent à hauteur de 17.9% du CA total et payent majoritairement par carte de crédit (72.7%) et par boleto (20.5%). Leurs catégories préférées sont "bed bath table" (10.3%), "sports leisure" (8.4%) et "health beauty" (8.0%). Ces clients ont passé en moyenne 1.0 commandes dont la dernière a eu lieu il y a en moyenne 257.2 jours à la date de l'analyse. Ces clients proviennent à 41.8% de l'état "SP". Ils ont donné un score moyen de 4.1 étoiles.
3. Le ***cluster 2*** représentent 0.3% des clients. En moyenne, ils achètent 1.77 articles au prix moyen de 1372.58, soit un panier moyen de 2429.46. Ils participent à hauteur de 3.9% du CA total et payent majoritairement par carte de crédit (77.7%) et par boleto (18.2%). Leurs catégories préférées sont "computers accessories" (10.1%), "musical instruments" (9.4%) et "auto" (9.0%). Ces clients ont passé en moyenne 1.1 commandes dont la dernière a eu lieu il y a en moyenne 230.1 jours à la date de l'analyse. Ces clients proviennent à 33.2% de l'état "SP". Ils ont donné un score moyen de 3.9 étoiles.
4. Le ***cluster 3*** représentent 11.1% des clients. En moyenne, ils achètent 1.43 articles au prix moyen de 216.46, soit un panier moyen de 309.54. Ils participent à hauteur de 20.8% du CA total et payent majoritairement par carte de crédit (79.3%) et par boleto (15.6%). Leurs catégories préférées sont "bed bath table" (9.7%), "health beauty" (9.1%) et "watches gifts" (8.3%). Ces clients ont passé en moyenne 1.1 commandes dont la dernière a eu lieu il y a en moyenne 185.5 jours à la date de l'analyse. Ces clients proviennent à 36.0% de l'état "SP". Ils ont donné un score moyen de 4.0 étoiles.
5. Le ***cluster 4*** représentent 31.8% des clients. En moyenne, ils achètent 1.11 articles au prix moyen de 91.85, soit un panier moyen de 101.95. Ils participent à hauteur de 19.7% du CA total et payent majoritairement par carte de crédit (74.3%) et par boleto (18.5%). Leurs catégories préférées sont "health beauty" (10.4%), "bed bath table" (9.9%) et "housewares" (7.2%). Ces clients ont passé en moyenne 1.0 commandes dont la dernière a eu lieu il y a en moyenne 84.8 jours à la date de l'analyse. Ces clients proviennent à 47.3% de l'état "SP". Ils ont donné un score moyen de 4.3 étoiles.
6. Le ***cluster 5*** représentent 21.4% des clients. En moyenne, ils achètent 1.14 articles au prix moyen de 105.94, soit un panier moyen de 120.77. Ils participent à hauteur de 15.7% du CA total et payent majoritairement par carte de crédit (71.3%) et par boleto (20.7%). Leurs catégories préférées sont "bed bath table" (10.6%), "furniture decor" (8.3%) et "sports leisure" (8.0%). Ces clients ont passé en moyenne 1.0 commandes dont la dernière a eu lieu il y a en moyenne 460.3 jours à la date de l'analyse. Ces clients proviennent à 39.2% de l'état "SP". Ils ont donné un score moyen de 4.2 étoiles.
7. Le ***cluster 6*** représentent 0.0% des clients. En moyenne, ils achètent 3.29 articles au prix moyen de 2395.58, soit un panier moyen de 7881.47. Ils participent à hauteur de 0.4% du CA total et payent majoritairement par carte de crédit (50.0%) et par boleto (50.0%). Leurs catégories préférées sont "fixed telephony" (25.0%), "small appliances" (25.0%) et "agro industry and commerce" (12.5%). Ces clients ont passé en moyenne 1.1 commandes dont la dernière a eu lieu il y a en moyenne 318.9 jours à la date de l'analyse. Ces clients proviennent à 28.6% de l'état "ES". Ils ont donné un score moyen de 2.7 étoiles.
8. Le ***cluster 7*** représentent 1.0% des clients. En moyenne, ils achètent 1.63 articles au prix moyen de 787.18, soit un panier moyen de 1283.11. Ils participent à hauteur de 7.7% du CA total et payent majoritairement par carte de crédit (78.9%) et par boleto (15.3%). Leurs catégories préférées sont "watches gifts" (11.5%), "health beauty" (9.3%) et "computers" (8.5%). Ces clients ont passé en moyenne 1.1 commandes dont la dernière a eu lieu il y a en moyenne 240.8 jours à la date de l'analyse. Ces clients proviennent à 30.6% de l'état "SP". Ils ont donné un score moyen de 3.9 étoiles.